File: python_sdk/src/reality_capture/specifications/clearance_checker.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class ClearanceCheckerInput(BaseModel):
    model_3d: str = Field(alias="model3D",
                          description="Reality data id of a 3D model (mesh, point cloud, a ContextScene with embedded 3D model) to process")
    footprints: str = Field(alias="footprints",
                            description="Reality data id of ContextScene, annotated with embedded 3D footprints")


class ClearanceCheckerOutputs(BaseModel):
    clearance: Optional[str] = Field(None, alias="clearance",
                                     description="Reality data id of ContextScene, annotated with embedded 3D clearance")


class ClearanceCheckerOutputsCreate(Enum):
    CLEARANCE = "clearance"


class ClearanceCheckerSpecificationsCreate(BaseModel):
    inputs: ClearanceCheckerInput = Field(description="Inputs")
    outputs: list[ClearanceCheckerOutputsCreate] = Field(description="Outputs")


class ClearanceCheckerSpecifications(BaseModel):
    inputs: ClearanceCheckerInput = Field(description="Inputs")
    outputs: ClearanceCheckerOutputs = Field(description="Outputs")

File: typescript/packages/reality-capture/src/specifications/clearance_checker.ts
import { z } from "zod";

export const ClearanceCheckerInputsSchema = z.object({
  model3D: z.string().describe("Reality data id of a 3D model (mesh, point cloud, a ContextScene with embedded 3D model) to process"),
  footprints: z.string().describe("Reality data id of ContextScene, annotated with embedded 3D footprints"),
});
export type ClearanceCheckerInputs = z.infer<typeof ClearanceCheckerInputsSchema>;

export const ClearanceCheckerOutputsSchema = z.object({
  clearance: z.string().optional().describe("Reality data id of ContextScene, annotated with embedded 3D clearance"),
});
export type ClearanceCheckerOutputs = z.infer<typeof ClearanceCheckerOutputsSchema>;

export enum ClearanceCheckerOutputsCreate {
  CLEARANCE = "clearance",
}

export const ClearanceCheckerSpecificationsCreateSchema = z.object({
  inputs: ClearanceCheckerInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(ClearanceCheckerOutputsCreate)).describe("Outputs"),
});
export type ClearanceCheckerSpecificationsCreate = z.infer<typeof ClearanceCheckerSpecificationsCreateSchema>;

export const ClearanceCheckerSpecificationsSchema = z.object({
  inputs: ClearanceCheckerInputsSchema.describe("Inputs"),
  outputs: ClearanceCheckerOutputsSchema.describe("Outputs"),
});
export type ClearanceCheckerSpecifications = z.infer<typeof ClearanceCheckerSpecificationsSchema>;